src/devices/bus/isa/sblaster_lle.cpp

```diff
@@ -52,12 +53,27 @@ void isa8_sblaster_2_0_lle_device::device_add_mconfig(machine_config &config)
 
 	PC_JOY(config, m_joy);
 
-	SPEAKER(config, m_speaker).front_center();
-	MC1408(config, m_dac, 0).add_route(ALL_OUTPUTS, m_speaker, 0.5);
+	SPEAKER(config, m_lspeaker).front_left();
+	SPEAKER(config, m_rspeaker).front_right();
+
+	MC1408(config, m_dac, 0);
+	m_dac->add_route(ALL_OUTPUTS, m_lspeaker, 0.5);
+	m_dac->add_route(ALL_OUTPUTS, m_rspeaker, 0.5);
 
 	YM3812(config, m_ym3812, CLOCK_YM3812);
-	m_ym3812->add_route(ALL_OUTPUTS, m_speaker, 1.0);
-	/* no CMS */
+	m_ym3812->add_route(ALL_OUTPUTS, m_lspeaker, 1.0);
+	m_ym3812->add_route(ALL_OUTPUTS, m_rspeaker, 1.0);
+
+	if(m_cms_present)
+	{
+		// FM and DAC are mono, but CMS is stereo.
+		SAA1099(config, m_saa1099_1, CLOCK_SAA1099);
+		m_saa1099_1->add_route(0, m_lspeaker, 0.5);
+		m_saa1099_1->add_route(1, m_rspeaker, 0.5);
+		SAA1099(config, m_saa1099_2, CLOCK_SAA1099);
+		m_saa1099_2->add_route(0, m_lspeaker, 0.5);
+		m_saa1099_2->add_route(1, m_rspeaker, 0.5);
+	}
```

Comment:
This can’t possibly work.  Machine config happens before device start/reset, and can’t depend on the state of I/O ports anyway (and if it could, it would wreak havoc with stuff like `-listxml` and `-listroms`).

Reply:
Hmm, the CMS is independent of everything else on the board, so I can leave it not hooked up if the machine config switch is disabled on a hard reset.